src/main/java/org/cloudfoundry/promregator/auth/NullEndpointProvider.java

```diff
@@ -0,0 +1,40 @@
+package org.cloudfoundry.promregator.auth;
+
+import static com.sap.cloud.security.xsuaa.Assertions.assertNotNull;
+
+import java.net.URI;
+
+import com.sap.cloud.security.config.CredentialType;
+import com.sap.cloud.security.config.OAuth2ServiceConfiguration;
+import com.sap.cloud.security.xsuaa.client.OAuth2ServiceEndpointsProvider;
+
+public class NullEndpointProvider implements OAuth2ServiceEndpointsProvider {
+
+	private final URI baseUri;
+	private final URI certUri;
+
+	public NullEndpointProvider(OAuth2ServiceConfiguration config) {
```

Comment:
Stupid question: why of type `OAuth2ServiceConfiguration` and not using `OAuth2XSUAAAuthenticationConfiguration` directly?

Reply:
The reason is: We are re-using the  `OAuth2ServiceConfiguration#getCredentialType()`. This is most likely used also otherwise inside the token-client. So we are making the decision in the same way. I think we should keep it like it is.